assets/src/edit-story/components/panels/document/publish/publishTime.js

```diff
@@ -106,14 +113,18 @@ function PublishTime() {
         placement={PLACEMENT.BOTTOM_END}
         renderContents={({ propagateDimensionChange }) => (
           <DateTime
-            value={date}
+            value={floatingDate ? Date.now() : date}
```

Comment:
Any issues when the site is set to use different timezones?

Reply:
Actually, there are already currently existing issues with the site set to use a different timezone. I'll create a separate issue for looking into that.
To repro:
- Set the time to something
- Save and refresh, see the time has changed.

Reply:
Added: https://github.com/google/web-stories-wp/issues/7780